### PR TITLE
STAC / OGC collections: implement RangeType schema

### LINKED ISSUE #916: STAC: support domain and range
**Describe the bug**

The OGC STAC / collections API doesn't provide domain set grid data or a range type schema. The grid can be defined by extending the existing `extent` object in the collection description. Example:

```json
  "extent" : {
      "spatial" : {
         "bbox" : [ [ -180, -90, 180, 90 ] ],
         "grid" : [
            {
               "cellsCount" : 1024,
               "resolution" : 0.3515625
            },
            {
               "cellsCount" : 512,
               "resolution" : 0.3515625
            }
         ]
      },
      "temporal" : {
         "interval" : [ [ "2015-12-30", "2025-12-26" ] ],
         "grid" : {
            "cellsCount" : 3650,
            "resolution" : "P1D"
         }
      },
      "pressure" : {
         "crs" : "http://www.opengis.net/def/crs/OGC/0/barometric",
         "interval" : [
            [ 1000.0, 10.0 ]
         ],
         "grid" : {
            "cellsCount" : 8,
            "coordinates" : [ 1000.0, 850.0, 700.0, 500.0, 250.0, 100.0, 50.0, 10.0 ]
         }
      }
   },
```

The range type should be provided as a JSON schema at an additional `/schema` endpoint under the collection identifier. Example:

```json
{
   "$schema" : "https://json-schema.org/draft/2020-12/schema",
   "$id" : "https://maps.gnosis.earth/ogcapi/collections/windSpeed/schema",
   "title" : "Wind Velocity",
   "type" : "object",
   "properties" : { "ua" : {
      "title" : "Eastward Wind",
      "type" : "number",
      "x-ogc-property-seq" : 1
   }, "va" : {
      "title" : "Northward Wind",
      "type" : "number",
      "x-ogc-property-seq" : 2
   } }
}

```

### PR DESCRIPTION
**Pull request by pont-us**
Closes #916

* Add spatial and temporal grid data to collection descriptions
* Add a schema endpoint returning a JSON schema of a dataset's data variables
* Add links to domain set, range type, and range schema to collection descriptions
* Exclude grid mapping variables when returning data variable information via OGC range type and schema endpoints

Note: the auto-generated Codecov comment below may not be up to date; click through to the full Codecov report to confirm that this PR has full test coverage.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: test/webapi/ows/coverages/test_controllers.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
     get_dataarray_description,
     get_units,
     is_xy_order,
-    transform_bbox,
+    transform_bbox, get_coverage_rangetype_for_dataset,
```
Comment (TonioF) [inline test/webapi/ows/coverages/test_controllers.py:42]:
```suggestion
    transform_bbox, 
    get_coverage_rangetype_for_dataset,
```

File: test/webapi/ows/stac/test_controllers.py [hunk 2]
```diff
@@ -44,15 +45,17 @@
     STAC_VERSION,
     get_collection_queryables,
     get_datacube_dimensions,
-    get_single_collection_items, 
-    crs_to_uri_or_wkt,
+    get_single_collection_items,
+    crs_to_uri_or_wkt, get_time_grid,
```
Comment (TonioF) [inline test/webapi/ows/stac/test_controllers.py:49]:
```suggestion
    crs_to_uri_or_wkt, 
    get_time_grid,
```

File: xcube/webapi/ows/stac/routes.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 from .api import api
 from .context import StacContext
 from .controllers import get_collection, get_collection_queryables, \
-    get_single_collection_items
+    get_single_collection_items, get_collection_schema
```
Comment (TonioF) [inline xcube/webapi/ows/stac/routes.py:27]:
```suggestion
    get_single_collection_items, 
    get_collection_schema
```

**Review by TonioF — APPROVED** [review]
Overall, this is looking really good. I only have three minor design suggestions.

**Comment by pont-us** [comment]
> Overall, this is looking really good. I only have three minor design suggestions.

Thanks! All suggestions applied.

File: xcube/webapi/ows/stac/controllers.py [hunk 4]
```diff
@@ -385,6 +386,51 @@ def get_collection_queryables(
...
+
+    ml_dataset = ctx.get_ml_dataset(collection_id)
+    ds = ml_dataset.base_dataset
+
+    def get_title(var_name: str) -> str:
+        attrs = ds[var_name].attrs
+        return attrs['long_name'] if 'long_name' in attrs else var_name
+
+    return {
+        '$schema': 'https://json-schema.org/draft/2020-12/schema',
```
Comment (thomasstorm) [inline xcube/webapi/ows/stac/controllers.py:417]: Maybe put this string into a constant.

File: xcube/webapi/ows/stac/controllers.py [hunk 5]
```diff
@@ -532,6 +588,29 @@ def _get_single_dataset_collection(
...
+                'type': 'application/json',
+                'title': 'Range type of the coverage',
+            },
+            {
+                'rel': 'http://www.opengis.net/def/rel/ogc/1.0/'
+                       'coverage-domainset',
+                'href': f'{base_url}{PATH_PREFIX}/collections/'
+                        f'{dataset_id}/coverage/domainset?f=json',
+                'type': 'application/json',
+                'title': 'Domain set of the coverage',
```
Comment (thomasstorm) [inline xcube/webapi/ows/stac/controllers.py:612]: Same here, maybe move the URL strings into constants?

**Review by thomasstorm — APPROVED** [review]
Looking good! Two minor suggestions, but approved nonetheless.

File: xcube/webapi/ows/stac/controllers.py (hunk 4 above)
Comment (pont-us) [reply xcube/webapi/ows/stac/controllers.py]: I don't see much need, but don't really have anything against it either, so done.

File: xcube/webapi/ows/stac/controllers.py (hunk 5 above)
Comment (pont-us) [reply xcube/webapi/ows/stac/controllers.py]: Done.

**Pull request merged**